src/pydata_sphinx_theme/assets/styles/abstracts/_links.scss

```diff
@@ -164,114 +179,54 @@ $link-hover-decoration-thickness: string.unquote(
   }
 }
 
-// Navigation bar current page link styles
-// ---------------------------------------
-// Adds a bottom underline, this leaves enough space for the hover state without
-// cluttering the navbar.
-// We want the side box shadow to have the same thickness as the hover underline
-@mixin link-navbar-current {
-  font-weight: 600;
-  color: var(--pst-color-primary);
-
-  @if $link-hover-decoration-thickness {
-    border-bottom: $link-hover-decoration-thickness
-      solid
-      var(--pst-color-primary);
-  }
-}
-
-// Navigation bar icon links hover styles
-// --------------------------------------
-// Adds a bottom box-shadow - since there is no text we cannot use text-decoration
-// We want the side box shadow to have the same thickness as the hover underline
-@mixin icon-navbar-hover {
-  &:hover {
-    color: var(--pst-color-link-hover);
-
-    @if $link-hover-decoration-thickness {
-      box-shadow: 0
-        $link-hover-decoration-thickness
-        0
-        var(--pst-color-link-hover);
-    }
-  }
-}
+// Heaver navbar text and icon links
+// ---------------------------------
+// (includes light/dark mode button)
+
+// This mixin makes it possible to show hover/underline and focus/ring styles at
+// the same time. The trick is to use:
+//    - a pseudo-element with bottom border for the hover underline
+//    - a CSS outline for the focus ring.
+
+// Normally we use box-shadow for underline and outline for focus ring. But we
+// cannot apply box-shadow and outline together on the same element because the
+// border-radius value that we use to round the outline will also round the
+// box-shadow used for the underline. We also cannot use text-underline because
+// it does not work on non-text links, nor do we want to use it on text links
+// that we want to treat as blocks, such as the header nav links because the
+// underline will wrap across two lines if the link text also wraps across two
+// lines.
+@mixin link-style-block {
+  color: var(--pst-color-text-muted);
 
-// Mixin for links in the header (and the More dropdown toggle).
+  // Set position relative so that the child ::before pseudo-element's absolute
+  // position is relative to this element.
+  position: relative;
 
-// The mixin assumes it will be applied to some element X with a markup structure
-// like: X > .nav-link, or X > .dropdown-toggle.
+  // Put some space between the underline and the content. Must apply to top as
+  // well as bottom so that the link is centered within the focus ring.
+  padding-block: 0.25rem;
 
-// It also assumes X.current is how the app annotates which item in the header nav
-// corresponds to the section in the docs that the user is currently reading.
-@mixin header-link {
```

Comment:
This mixin was only being used in one place, so I decided to rework it to be a bit more general and to unify the hover-and-focus styling shared between the navbar text links and the navbar icon links